lib/pbench/agent/tool_data_sink.py

```diff
@@ -381,216 +443,321 @@ def tm_log_capture(self):
+                    noop_tools.append(tool_name)
+                elif tool_name in transient_tools_l:
+                    transient_tools.append(tool_name)
+                else:
+                    self.logger.error(
+                        f"Registered tool {tool_name} is not recognized in tool metadata"
+                    )
+            tm["noop_tools"] = noop_tools
+            tm["persistent_tools"] = persistent_tools
+            tm["transient_tools"] = transient_tools
+
+            if tm["hostname"] == self.hostname:
+                # The "localhost" tool meister instance does not send data
+                # to the tool data sink, it just writes it locally.
+                tm["posted"] = None
+            elif not transient_tools:
+                # Only Tool Meisters with at least one transient tool will
+                # send data to a data sink, so ignore those Tool Meisters
+                # without any.
+                tm["posted"] = None
+            else:
+                # The `posted` field is "dormant" to start (as set below),
+                # "waiting" when we transition to the "send" state, "dormant"
+                # when we receive data from the target Tool Meister host.
+                tm["posted"] = "dormant"
+
+        # Opportunistically capture the caller's SSH configuration in case
+        # they have host specific configurations they might want to consider
+        # in the future when reviewing historical data.  The host names used
+        # in the config file might be very different from what tools capture
+        # in their output.
+
+        # cp -L  ${HOME}/.ssh/config   ${dir}/ssh.config > /dev/null 2>&1
+        home = os.environ.get("HOME", "")
+        if home:
+            src = str(Path(home) / ".ssh" / "config")
+            dst = str(self.benchmark_run_dir / "ssh.config")
+            try:
+                shutil.copyfile(src, dst)
+            except FileNotFoundError:
+                pass
+            except Exception as exc:
+                self.logger.warning("failed to copy %s to %s: %s", src, dst, exc)
+        # cp -L  /etc/ssh/ssh_config   ${dir}/ > /dev/null 2>&1
+        etc_ssh = Path("/etc") / "ssh"
+        src = str(etc_ssh / "ssh_config")
+        dst = str(self.benchmark_run_dir / "ssh_config")
         try:
-            self.tm_log_capture_thread.join()
-        except Exception:
+            shutil.copyfile(src, dst)
+        except FileNotFoundError:
             pass
-        if self._tm_logging_pubsub:
-            self.logger.debug("unsubscribe tm-logging")
-            self._tm_logging_pubsub.unsubscribe()
-            self.logger.debug("pubsub close tm-logging")
-            self._tm_logging_pubsub.close()
+        except Exception as exc:
+            self.logger.warning("failed to copy %s to %s: %s", src, dst, exc)
+        # cp -rL /etc/ssh/ssh_config.d ${dir}/ > /dev/null 2>&1
+        subprocess.run(
```

Comment:
We need a comment here about why we are using a spawned `cp -rL` command instead of `shutil.copytree()`.